tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages.Tests/NuGetExtensionsTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using Xunit;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Packages.Tests
+{
+    public class NuGetExtensionsTests
+    {
+        [Fact]
+        public void GetNuGetVersionThrowsWhenNull()
+        {
+            NuGetReference? reference = default;
+
+            Assert.Throws<ArgumentNullException>(() => reference!.GetNuGetVersion());
+        }
+
+        [InlineData("6.0.*")]
+        [InlineData("4.*")]
+        [InlineData("*")]
+        [Theory]
+        public void GetNuGetVersionCanHandleFloatingVersions(string version)
```

Comment:
What about passing in here the expected version? I'd rather be explicit here in the test about expected in and out rather than logic to create the expected